### ISSUE TITLE
Live Unit Testing excludes class from tests when modifying code

### ISSUE THREAD

**Issue by Gibsouille** [nunit/nunit3-vs-adapter#525]
When I launch my project, Live Unit Testing (LUT) automatically starts and runs all the NUnit tests successfully, but when I modify some code it excludes tests about the code I have just modified. 

Live Unit Testing verbose logs tell me those bad words : "[10:01:16.174 Error] - TestDriver - [TestRunner 1] Une exception s'est produite durant l'appel de l'exécuteur 'executor://nunit3testexecutor/' : Format incorrect pour le TestCaseFilter Opérateur manquant '|' ou '&'. Spécifiez le format approprié, puis réessayez. Notez que si le format est incorrect, aucun test n'est exécuté."
Sorry for the french but even after changing VS language to english I still get the error in french.

Running the tests through TestExplorer->RunAll does run all the tests, but I still get the LUT "excluded" icon in the side bar and LUT doesn't execute the tests it has excluded. 

Got VS 2017 v15.7.3 (working with .NET 4.5.2), and NUnit 3.6.1 with NUnit3TestAdapter 3.7.0. 
Tried to update NUnit and adapter to latest stable 3.10.0 but it didn't work better. 

Help please! LUT isn't really interesting with that problem :)

**Comment by DinoSourcesRex** [comment]
@OsirisTerje I am getting the same issue. It suddenly started occurring in a project that was otherwise fine. If I delete the `lut` folder in the `.vs` folder then it works for the first run of tests, then goes back to failing.

I can't share the code as it's production code unfortunately. 

```
[13:56:11.844 Error] [TestRunner 1] An exception occurred while invoking executor 'executor://nunit3testexecutor/': Incorrect format for TestCaseFilter Missing Operator '|' or '&'. Specify the correct format and try again. Note that the incorrect format can lead to no test getting executed.
```

You mentioned `runsettings` above, where can I see / edit these settings as I have never touched them. Everything I see online points to command line arguments.

Perhaps worth nothing that running through the test explorer, manually, works fine.

---
Visual Studio Enterprise 2017 v15.7.4
.NET Core 2.1
AutoFixture 4.4.0
FluentAssertions 5.3.0
Microsoft.NET.Test.Sdk 15.7.2
NSubstitute 3.1.0
NUnit 3.10.1
NUnit3TestAdapter 3.10.0

**Comment by OsirisTerje** [comment]
@DinoSourcesRex You add a runsettings file manually.  You can use the [item template for it](https://marketplace.visualstudio.com/items?itemName=OsirisTerje.Runsettings-19151) . It will then override any default stuff they have there.  
And, you might need to add the information about the runsettings file to the dotnet runner.  

I'll try to create a small 2.1 repro project for this, but the one I have had earlier have all worked, anyway, I'll try to check again.  If you could create a small one that mimic what you see, it would be awesome.

Is there anything special with your tests ?   (which is why it would be great if you wrote the mimic/repro project)

**Comment by DinoSourcesRex** [comment]
@OsirisTerje Nothing special, no. I'm just using some `TestCase`'s and regular Tests. Also got `SetUp` in there too, but that's all the functionality I utilise. The TestCase inputs do contain `/ , . -` though, maybe that's what is the problem?

**Comment by DinoSourcesRex** [comment]
@OsirisTerje The setup at the time was utilising Specflow, which doesn't support netcore yet so the testing project was set to version 472, rather than having compatibility issues I switched my main application into a `netstandard2.0` library and a `netcoreapp2.0` for the api (which references the library).

---

Regarding the problem, I switched test frameworks to see if another framework would have an issue running the code and it didn't so I simply stayed with that framework. 

I've created a small repro that should let you see the problem. Look in the `TestProj/Infrastructure/Extensions/StringExtension.cs` class, I left some notes there.

https://github.com/DinoSourcesRex/NunitRepro

**Comment by OsirisTerje** [comment]
Yes, I can verify the behaviour.  I didnt find any test there that actually used the StringExtension.cs class (guess you forgot to commit that?)  , but added that and then could observe what you mention.  I don't need to clean out the directory though, if I just include the test method again, LUT starts up again.  Another edit, just adding a blank line, and then it seems to exclude the test again.   
![image](https://user-images.githubusercontent.com/203432/43548166-9a9a4820-95dd-11e8-9400-45a063626277.png)

Including the test again:
![image](https://user-images.githubusercontent.com/203432/43548331-f34e4ed0-95dd-11e8-80a5-0e435e9b628e.png)

And it repeats and repro consistently. 

The code to be changed is just this (in method mentioned above by @DinoSourcesRex )
When it works:
![image](https://user-images.githubusercontent.com/203432/43548513-5f5c296c-95de-11e8-829c-19e29ff222c2.png)
Then just delete the blank line (red arrow), and it goes:
![image](https://user-images.githubusercontent.com/203432/43548536-71499ad8-95de-11e8-832a-be081f9467d7.png)



I don't think this is the adapter, it more looks like a LUT  failure.  I'll bring this to the attention of the product group handling that.  

I have added a PR for the test, see https://github.com/DinoSourcesRex/NunitRepro/pull/1 

My fork  with the test is at: https://github.com/OsirisTerje/NunitRepro


Thanks for bringing up this!

**Comment by DinoSourcesRex** [comment]
Well that's embarrassing haha, but at least you managed to get something useful for it. I took the live code and left the only thing I felt comfortable leaving. I was sure there should have been 4 tests in there but...ah well!

I'll leave the repo up until until I forget why I left it up, then I'll delete it. I'm sure that gives us a good few months.

**Comment by genlu** [comment]
Hi folks, thanks for reporting this issue. We have fixed this in VS 2017 15.8 Preview 3 (released in June). However, because this is a .NET Core test project, in addition to upgrade VS, you will also need to upgrade the reference of `Microsoft.NET.Test.Sdk` NuGet package to latest (15.8.0) to pick up the fix. For .NET framework test project this step is not necessary because it will use the bits shipped with VS.

**Comment by OsirisTerje** [comment]
Thanks @genlu !!
